### PR TITLE
Add Strict entity

### PR DESCRIPTION
**Pull request by bplommer**
I've merged `Empty` into `Strict` to avoid having two legal representations for a strict empty entity - any thoughts on this?

### CODE REVIEW

File: core/shared/src/main/scala/org/http4s/Entity.scala
Comment (armanbilge) [inline core/shared/src/main/scala/org/http4s/Entity.scala:54]: Bikeshedding time :) I'm of the opinion we should use a scodec `ByteVector` here rather than a `Chunk`. Is there a reason you felt drawn to `Chunk`?

**Comment by armanbilge** [comment]
So, regarding the `Empty` vs `Strict` thing, I do agree it would be nice to avoid having two legal representations for this. Also it opens the question whether a distinct `Empty` entity should extend `Strict`. I wonder if we can reach for a representation like `sealed trait Strict` inhabited by `EmptyStrict` and `NonEmptyStrict`, where the non-empty one behaves like any non-empty collection in cats. That way we can have real static guarantees about this stuff, which might be useful (?) Like maybe too crazy, but I'm reminded of your PRs using singleton types to be ultra expressive, such statically requiring a 204 response to be paired with `EmptyStrict` type, and that entity only.

As always, thanks for keeping this series of PRs going!! :)

File: core/shared/src/main/scala/org/http4s/Entity.scala
Comment (bplommer) [reply core/shared/src/main/scala/org/http4s/Entity.scala:54]: Nope, just copying what @ChristopherDavenport did in the original PR. I'm totally open to `ByteVector` and see the appeal of not using an fs2 type here.

Comment (armanbilge) [reply core/shared/src/main/scala/org/http4s/Entity.scala:54]: Cool. My vote is for `ByteVector` then, curious to hear other's thoughts on this as well. My thinking is along the lines of:
1. `Chunk` is almost an fs2 internal thing here, as you point out
2. `ByteVector` is a lot more optimized for `Byte` than `Chunk[Byte]`
3. Precedent/defer to authority in https://github.com/http4s/http4s/pull/5321#discussion_r721641372

**Comment by bplommer** [comment]
@armanbilge I did try something like that but then I started bikeshedding with myself so decided to just open the simplest possible PR. Maybe I'll give it another go though 😄

**Comment by bplommer** [comment]
Following discussion on Discord, I'm keeping `Empty` as a separate case and leaving the representation as `Chunk` for now.

File: core/shared/src/main/scala/org/http4s/Entity.scala
Comment (armanbilge) [reply core/shared/src/main/scala/org/http4s/Entity.scala:51]: Ok, the consensus on Discord is for `Chunk` primarily b/c `Chunk` can subsume `ByteVector` (but not the other way around), so this would let us avoid unnecessary copying.

Personally I still like `ByteVector` for the reasons above, but `Chunk[Byte]` is good too.

**Review by armanbilge — APPROVED** [review]

**Comment by armanbilge** [comment]
Hum, I thought we'd wait a little more before merging. I remember @danicheg might have been deferring some discussions for this PR.

**Comment by bplommer** [comment]
Good point, sorry for being so trigger-happy 😬 I'll revert it and open a new one.

**Pull request merged**
